submissions/semenjob/memoryCards/game-logic.js

```diff
@@ -0,0 +1,152 @@
+"use strict";
+import { memoryCardImages, memoryCardFrontImage } from "./constant.js";
+
+document.addEventListener("DOMContentLoaded", (event) => {
+  startGame();
+});
+
+const startGameBtn = document.querySelector(".start");
+const resetGame = document.querySelector(".reset");
+
+const toggleFlip = (card) => {
+  if (card !== null) {
+    card.classList.toggle("flip");
+    card.classList.toggle("block");
+  }
+  return;
+};
+
+const initMemoryCardIndexes = (n) => {
+  return Array(n)
+    .fill()
+    .map((_, idx) => 0 + idx)
+    .flatMap((num) => [num, num])
+    .sort(() => Math.random() - 0.5);
+};
+
+function startGame() {
+  const createMemoryCard = (id, front, back) => {
+    const flipperTemplate = `
+    <div class="flipper">
+      <div class="front-view"><img src="${front}" alt="Card front view"></div>
+      <div class="back-view"><img src="${back.img}" alt="${back.alt}"></div>
+    </div>
+    `;
+
+    const card = document.createElement("div");
+    card.classList.add("memory_card", "block");
+    card.innerHTML = flipperTemplate;
+    card.dataset.cardId = id;
+    return card;
+  };
+
+  const cards = document.querySelector(".memory_cards");
+  const cardsLength = 6;
+  const cardIndexes = initMemoryCardIndexes(cardsLength);
+
+  for (let i = 0; i < cardIndexes.length; i++) {
+    const cardId = cardIndexes[i];
+    const backgroundImage = memoryCardImages[cardId];
+    const card = createMemoryCard(
+      cardId,
+      memoryCardFrontImage,
+      backgroundImage
+    );
+    cards.append(card);
```

Comment:
At this line, you append a card on each iteration, avoid that pattern. 
Use `append` method, which can add several elements in one operation